torch/distributed/optim/zero_redundancy_optimizer.py

```diff
@@ -147,23 +145,41 @@ def __init__(
         parameters_as_bucket_view: bool = False,
         **default: Any,
     ):
+        # Validate the type and length of `params`
+        # NOTE: While `Optimizer` allows `params` to be an iterable of `dict`s,
+        # ZeRO strictly requires `params` to be an iterable of `torch.Tensor`s
+        if (isinstance(params, torch.Tensor)):
+            raise TypeError("params argument should be an iterable of "
+                            "Tensors, but got " + torch.typename(params))
+        try:
+            self._all_params = list(params)
+        except TypeError:
+            raise TypeError("params argument should be an iterable of "
+                            "Tensors, but got " + torch.typename(params))
+        if (len(self._all_params) == 0):
+            raise ValueError("ZeroRedundancyOptimizer got an empty parameter "
```

Comment:
curious, why this is an error? Does local optimizer consider this as an error?

Reply:
Yes, this was following the local optimizer implementation. See [here](https://github.com/pytorch/pytorch/blob/095cd6a0da255a5ed24d146898c53790f0e9c3f0/torch/optim/optimizer.py#L49).